fortranformat/_output.py

```diff
@@ -554,78 +492,70 @@ def _output_float(w, d, e, state, ft, buff, sign_bit, zero_flag, ndigits, edigit
     if (nblanks < 0) or (edigits == -1):
         return '*' * w
     # See if we have space for a zero before the decimal point
-    if (nbefore == 0) and (nblanks > 0):
-        leadzero = True
+    leadzero = (nbefore == 0) and (nblanks > 0)
+    if leadzero:
         nblanks = nblanks - 1
-    else:
-        leadzero = False
-    out = ''
+
+    out = StringIO()
     # Pad to full field width
     if (nblanks > 0) and not PROC_NO_LEADING_BLANK:  # dtp->u.p.no_leading_blank
-        out = out + ' ' * nblanks
+        out.write(' ' * nblanks)
     # Attach the sign
-    out = out + sign
+    out.write(sign)
     # Add the lead zero if necessary
     if leadzero:
-        out = out + '0'
+        out.write('0')
     # Output portion before the decimal point padded with zeros
     if nbefore > 0:
         if nbefore > ndigits:
-            out = out + digits[:ndigits] + (' ' * (nbefore - ndigits))
+            out.write(digits[:ndigits] + (' ' * (nbefore - ndigits)))
             digits = digits[ndigits:]
             ndigits = 0
         else:
             i = nbefore
-            out = out + digits[:i]
+            out.write(digits[:i])
             digits = digits[i:]
             ndigits = ndigits - i
+
     # Output the decimal point
-    out = out + PROC_DECIMAL_CHAR
+    out.write(PROC_DECIMAL_CHAR)
+
     # Output the leading zeros after the decimal point
     if nzero > 0:
-        out = out + ('0' * nzero)
+        out.write(('0' * nzero))
+
     # Output the digits after the decimal point, padded with zeros
     if nafter > 0:
-        if nafter > ndigits:
-            i = ndigits
-        else:
-            i = nafter
-        zeros = '0' * (nafter - i)
-        out = out + digits[:i] + zeros
-        digits = digits[nafter:]
-        ndigits = ndigits - nafter

```

Comment:
I think we missed an update to `ndigits` here - this should break a test unless we no longer need it?

Reply:
`ndigits` isn't actually used after this point, so it's safe to delete -- the tests also do still pass